autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java

```diff
@@ -43,16 +52,56 @@ public GovernorImpl(final int stageOneLimit, final int stageTwoLimit, final int
         this.stageOneShutdownPriorityLimit = stageOneLimit;
         this.stageTwoShutdownPriorityLimit = stageTwoLimit;
         this.stageThreeShutdownPriorityLimit = stageThreeLimit;
+        this.scalerThreads = new HashMap<>();
+    }
+
+    @Override
+    public boolean makeRoom(final String serviceRef) throws ScalerException
+    {
+        final AdvancedInstanceInfo lastInstanceInfo = instanceInfoMap.getOrDefault(serviceRef, null);
+        if (lastInstanceInfo == null) {
+            return false;
+        }
+        final double percentageDifference = lastInstanceInfo.getPercentageDifference();
+        final Map<String, AdvancedInstanceInfo> possibleVictims = instanceInfoMap.entrySet().stream()
+            .filter(e -> e.getValue().getPercentageDifference() > -1)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        String selectedVictim = "";
+        for (final Map.Entry<String, AdvancedInstanceInfo> possibleVictim : possibleVictims.entrySet()) {
+            if (isStringNullOrEmpty(selectedVictim) || possibleVictim.getValue().getPercentageDifference()
+                < instanceInfoMap.getOrDefault(selectedVictim, null).getPercentageDifference()) {
+                if (percentageDifference != -1) {
+                    if (possibleVictim.getValue().getPercentageDifference() > percentageDifference
+                        || scalingConfigurationMap.get(possibleVictim.getKey()).getMinInstances()
+                        == possibleVictim.getValue().getTotalRunningAndStageInstances()) {
+                        continue;
```

Comment:
I'm finding this difficult to follow.  You're going to have to talk me through it.